source/common/header_source_impl.cc

```diff
@@ -0,0 +1,24 @@
+#include "common/header_source_impl.h"
+
+#include "external/envoy/source/common/common/assert.h"
+
+namespace Nighthawk {
+namespace Client {
+
+StaticHeaderSourceImpl::StaticHeaderSourceImpl(Envoy::Http::HeaderMapPtr&& header,
+                                               const uint64_t max_yields)
+    : header_(std::move(header)), yields_left_(max_yields) {
+  RELEASE_ASSERT(header_ != nullptr, "header can't equal nullptr");
+}
+
+HeaderGenerator StaticHeaderSourceImpl::get() {
+  return [this]() -> HeaderMapPtr {
+    while (yields_left_--) {
```

Comment:
s/if/while/ since we don't have "yeild" yet.  However, just give C++ some time to make http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/n4760.pdf official....

Reply:
I used to love that concept when I was working with C# a lot some ago, hope to see that soon in c++ :)